packages/server/src/server/handlers/agents.ts

```diff
@@ -195,16 +207,26 @@ export async function streamGenerateHandler({
       throw new HTTPException(404, { message: 'Agent not found' });
     }
 
-    const { messages, resourceId, resourceid, ...rest } = body;
+    const { messages, resourceId, resourceid, runtimeContext: agentRuntimeContext, ...rest } = body;
     // Use resourceId if provided, fall back to resourceid (deprecated)
     const finalResourceId = resourceId ?? resourceid;
+
+    const finalAgentRuntimeContext = agentRuntimeContext
+      ? new RuntimeContext(parse<Map<string, unknown>>(agentRuntimeContext))
+      : new RuntimeContext();
+
+    const finalRuntimeContext = new RuntimeContext<Record<string, unknown>>([
+      ...Array.from(runtimeContext.entries()),
+      ...Array.from(finalAgentRuntimeContext.entries()),
+    ]);
```

Comment:
The above means, you can optimize this one as well

```suggestion
    const finalRuntimeContext = new RuntimeContext<Record<string, unknown>>([
      ...Array.from(Object.entries(runtimeContext)),
      ...Array.from(agentRuntimeContext?.entries() ?? {}),
    ]);
```

Reply:
Yes, good changes, thanks.